File: Module/PlayWays/src/main/java-room-grab/com/module/playways/grab/room/view/RedPkgCountDownView.java
package com.module.playways.grab.room.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.RelativeLayout;

import com.module.playways.R;

public class RedPkgCountDownView extends RelativeLayout {
    RedCircleProgressView mRedCircleCountDownView;
    public RedPkgCountDownView(Context context) {
        super(context);
        init();
    }

    public RedPkgCountDownView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public RedPkgCountDownView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init(){
        inflate(getContext(), R.layout.red_pkg_count_down_view_layout, this);
        mRedCircleCountDownView = findViewById(R.id.red_circle_count_down_view);
    }

    public void startCountDown(long duration){
        mRedCircleCountDownView.go(duration);
    }
}

File: Module/PlayWays/src/main/java-room-grab/com/module/playways/grab/room/view/RedPkgCountDownView.kt
package com.module.playways.grab.room.view

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.RelativeLayout

import com.module.playways.R

class RedPkgCountDownView : RelativeLayout {
    private val mRedCircleCountDownView: RedCircleProgressView

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    init {
        View.inflate(context, R.layout.red_pkg_count_down_view_layout, this)
        mRedCircleCountDownView = findViewById(R.id.red_circle_count_down_view)
    }

    fun startCountDown(duration: Long) {
        mRedCircleCountDownView.go(duration)
    }
}
